arangod/Pregel/Worker.cpp

```diff
@@ -622,7 +639,13 @@ void Worker<V, E, M>::finalizeExecution(VPackSlice const& body,
     return;
   }
 
-  auto cleanup = [self = shared_from_this(), this, cb] {
+  VPackSlice store = body.get(Utils::storeResultsKey);
+  auto const doStore = store.isBool() && store.getBool() == true;
+  auto cleanup = [self = shared_from_this(), this, doStore, cb] {
+    if (doStore) {
+      _feature.metrics()->pregelWorkersStoringNumber->fetch_sub(1);
```

Comment:
I don't understand why you substract here first and then add later again in line 667 - should it not be the other way around?

Reply:
`cleanup` is a callback function that is called in lines 670 or after storing has finished, callback passed in in line 666;

Yes, this function is badly laid out. We should refactor it post freeze.

Reply:
makes sense, I did not see that!